### PR TITLE
Modify documentation and fix file extension in the test for AllenNLP integration.

### PR DESCRIPTION
**Pull request by himkt**
This PR modifies the documentation for AllenNLP integration.
And I also fix a small problem of incorrect specification for the file extension in the AllenNLP test.
(background: https://github.com/optuna/optuna/pull/1225#discussion_r427352698)

@toshihikoyanase @hvy 
Sorry for asking a review by name but could you please take a look?

### CODE REVIEW

File: optuna/integration/allennlp.py [hunk 1]
```diff
@@ -26,11 +26,13 @@ def dump_best_config(input_config_file: str, output_config_file: str, study: opt
 
     Args:
         input_config_file:
-            Input configuration file used with :class:`~optuna.integration.AllenNLPExecutor`.
+            Input configuration file (Jsonnet) specified in
+            :class:`~optuna.integration.AllenNLPExecutor`.
         output_config_file:
-            Output configuration file.
+            Output configuration (Json) file.
```
Comment (hvy) [inline optuna/integration/allennlp.py:32]: A nit but I think this is more common.
```suggestion
            Output configuration (JSON) file.
```

Comment (hvy) [reply optuna/integration/allennlp.py:32]: I'd personally remove the parenthesis here, mostly a matter of style.

File: optuna/integration/allennlp.py [hunk 2]
```diff
@@ -26,11 +26,13 @@ def dump_best_config(input_config_file: str, output_config_file: str, study: opt
...
         study:
-            An optimized study (``study.best_trial`` does not raise an error).
+            Instance of :class:`~optuna.study.Study`.
+            Note that :func:`~optuna.study.Study.optimize` mast have already called.
```
Comment (hvy) [inline optuna/integration/allennlp.py:35]: Nits.
```suggestion
            Note that :func:`~optuna.study.Study.optimize` must have been called.
```

File: optuna/integration/allennlp.py [hunk 3]
```diff
@@ -26,11 +26,13 @@ def dump_best_config(input_config_file: str, output_config_file: str, study: opt
 
     Args:
         input_config_file:
-            Input configuration file used with :class:`~optuna.integration.AllenNLPExecutor`.
+            Input configuration file (Jsonnet) specified in
+            :class:`~optuna.integration.AllenNLPExecutor`.
```
Comment (hvy) [inline optuna/integration/allennlp.py:30]: I think the previous "used with" sounds better.

**Comment by hvy** [comment]
I labeled this PR `document` rather than `test` since the former changes seem more significant.

File: optuna/integration/allennlp.py (hunk 2 above)
Comment (himkt) [reply optuna/integration/allennlp.py]: You're right, thank you.

File: optuna/integration/allennlp.py (hunk 3 above)
Comment (himkt) [reply optuna/integration/allennlp.py:30]: OK, I reverted this change.

**Comment by himkt** [comment]
@toshihikoyanase Thank you for the catch, I corrected it.

**Comment by himkt** [comment]
I have noticed [the documentation](https://42449-122299416-gh.circle-artifacts.com/0/docs/build/html/reference/integration.html?highlight=allennlp#optuna.integration.AllenNLPExecutor) and [the implementation](https://github.com/himkt/optuna/blob/modify-allennlp/optuna/integration/allennlp.py#L80-L88) do not match.
(The argument `include_package` only appears in the parameter section)

If someone knows what's wrong, please let me know.



Screen Shot 2020-05-24 at 16 59 49

**Comment by hvy** [comment]
I reproduced it in my environment too. 🤔

**Comment by hvy** [comment]
The issue seems to be caused by the experimental decorator. You can remove the decorate to observe that the documentation is rendered correctly.

**Pull request merged**